pyaedt/modules/MeshIcepak.py

```diff
@@ -1280,10 +1283,94 @@ def automatic_mesh_3D(self, accuracy, enable_stair_step=True):
         self.global_mesh_region.update()
         return True
 
+    @pyaedt_function_handler()
+    def assign_priorities(self, assignment):
+        """Set objects priorities.
+
+        Parameters
+        ----------
+        assignment : List[List[Union[Object3d, UserDefinedComponent, str]
+            List of lists of objects. Each list corresponds to one priority level from low to high.
+            This means that the first list has the lowest priority while the last list
+            has the highest priority. Objects not explicitly passed in the lists are assigned
+            to a priority level lower than the objects in the first list.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, "False" when failed.
+
+        References
+        ----------
+
+        >>> oEditor.UpdatePriorityList
+
+        Examples
+        --------
+
+        >>> ipk.mesh.assign_priorities([["Box1", "Rectangle1"], ["Box2", "Fan1_1"], ["Heatsink1_1"]])
+        """
+        if not isinstance(assignment[0], list):
+            raise AttributeError("``assignment`` input must be a list of lists.")
```

Comment:
```suggestion
        if not assignment or not isinstance(assignment, list) or not isinstance(assignment[0], list):
            raise AttributeError("``assignment`` input must be a list of lists.")
```
This should cover a wider rander of types for `assignment`. Currently, a `TypeError` will be raised if `assignment` is not subscriptable. If that's the behavior you wanted, please discard this comment.